api/src/main/java/org/openmrs/calculation/CalculationActivator.java

```diff
@@ -26,8 +31,41 @@
 
 	@Override
 	public void started() {
+		registerCalculations(
```

Comment:
@dkayiwa what's the relevance of that other ticket?

@wluyima, I _think_ things will work as I have them. It's definitely only necessary to update the token registrations at first startup (because these are part of implementation configuration that's outside of the OpenMRS lifecycle.

As the code is written, if the ImplementationConfiguredCalculationProvider.calculations cache is cleared, then the calculations will be reparsed when they are requested by name (via the token registration). So I think it's safe to leave this as-is. (Since refreshing the context will reload the ImplementationConfiguredCalculationProvider bean.)